/*
*  Copyright (c) 2020, The OpenThread Authors.
*  Copyright 2025 NXP
*  All rights reserved.
*
*  Redistribution and use in source and binary forms, with or without
*  modification, are permitted provided that the following conditions are met:
*  1. Redistributions of source code must retain the above copyright
*     notice, this list of conditions and the following disclaimer.
*  2. Redistributions in binary form must reproduce the above copyright
*     notice, this list of conditions and the following disclaimer in the
*     documentation and/or other materials provided with the distribution.
*  3. Neither the name of the copyright holder nor the
*     names of its contributors may be used to endorse or promote products
*     derived from this software without specific prior written permission.
*
*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
*  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
*  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
*  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
*  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
*  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
*  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
*  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
*  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
*  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
*  POSSIBILITY OF SUCH DAMAGE.
*/

/*
* Header for patch information.
* Description : This patch fixes a link error when building ot-daemon
*              tests/unit/ot-test-ncp-srp_server: Undefined reference to otChannelManagerRequestChannelSelect
*              otChannelManagerRequestChannelSelect is only implemented if OPENTHREAD_CONFIG_CHANNEL_MONITOR_ENABLE=1
* Version : v001
* Mandatory/Optional : Mandatory
* Impact if not applied : tests/unit/ot-test-ncp-srp_server can't be built
*/

Upstream-Status: Inappropriate  [platform specific changes]

diff --git a/src/ncp/ncp_base_ftd.cpp b/src/ncp/ncp_base_ftd.cpp
index cd1957ccb..d0a223b8c 100644
--- a/src/ncp/ncp_base_ftd.cpp
+++ b/src/ncp/ncp_base_ftd.cpp
@@ -1323,7 +1323,9 @@ template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_CHANNEL_MANAGER_CHANN
     otError error            = OT_ERROR_NONE;
 
     SuccessOrExit(error = mDecoder.ReadBool(skipQualityCheck));
+#if OPENTHREAD_CONFIG_CHANNEL_MONITOR_ENABLE
     error = otChannelManagerRequestChannelSelect(mInstance, skipQualityCheck);
+#endif
 
 exit:
     return error;
-- 
2.43.0

